client/dashboard/me/billing-history/dataviews.tsx

```diff
@@ -0,0 +1,383 @@
+import { sendReceiptEmailMutation } from '@automattic/api-queries';
+import { useMutation } from '@tanstack/react-query';
+import { Link, useNavigate } from '@tanstack/react-router';
+import { __experimentalText as Text, __experimentalVStack as VStack } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useMemo } from 'react';
+import { getReceiptUrlFor } from './urls';
+import {
+	formatReceiptAmount,
+	formatReceiptTaxAmount,
+	getTransactionTermLabel,
+	groupDomainProducts,
+	renderTransactionQuantitySummary,
+	summarizeReceiptItems,
+	transactionIncludesTax,
+} from './utils';
+import type { Receipt } from '@automattic/api-core';
+import type { Fields, Operator, SortDirection, View } from '@wordpress/dataviews';
+
+import './styles.scss';
+
+export const billingHistoryWideFields = [ 'date', 'service', 'type', 'amount' ];
+export const billingHistoryDesktopFields = [ 'date', 'service' ];
+export const billingHistoryMobileFields: string[] = [ 'service' ];
+
+export const defaultBillingHistoryView: View = {
+	type: 'table',
+	page: 1,
+	search: '',
+	perPage: 10,
+	fields: billingHistoryWideFields,
+	sort: {
+		field: 'date',
+		direction: 'desc' as SortDirection,
+	},
+	layout: {
+		styles: {
+			date: {
+				width: '14%',
+			},
+			service: {
+				width: '43%',
+			},
+			type: {
+				width: '20%',
+			},
+			amount: {
+				width: '23%',
+			},
+		},
+	},
+};
+
+export function useBillingHistoryActions() {
+	const navigate = useNavigate();
+	const sendEmailMutation = useMutation( {
+		...sendReceiptEmailMutation(),
+		meta: {
+			snackbar: {
+				success: __( 'Your receipt was sent by email successfully.' ),
+				error: __(
+					'There was a problem sending your receipt. Please try again later or contact support.'
+				),
+			},
+		},
+	} );
+
+	return useMemo(
+		() => [
+			{
+				id: 'view-receipt',
+				label: __( 'View receipt' ),
+				isEligible: ( item: Receipt ) => Boolean( item.id ),
+				callback: ( items: Receipt[] ) => {
+					const item = items[ 0 ];
+					navigate( {
+						to: getReceiptUrlFor( item ),
+					} );
```

Comment:
I just learned a more reliable way to do this in https://github.com/Automattic/wp-calypso/pull/106694 so it will work no matter what the base URL is.

```suggestion
					navigate( {
						to: receiptRoute.fullPath,
						params: { receiptId: item.id },
					} );
```

Reply:
Neat, thanks!